Open mSupply central (#28)
msupply-foundation/msupply_docs: content/docs/introduction/central.md

+++
title = "Open mSupply Central Server"
description = "Open mSupply central server functionality"
date = 2022-06-10T11:38:00+00:00
updated = 2022-06-10T11:38:00+00:00
draft = false

> weight = 8

Reviewer: <img width="239" alt="Screenshot 2024-04-24 at 15 28 54" src="https://github.com/msupply-foundation/msupply_docs/assets/61820074/5c664bc8-5fab-439e-8b41-ec6d8765b682">
Feels like this should be grouped by areas instead of alphabetical... Dashboard is 3rd from top and if you follow the navigation links at the bottom it doesn't make sense...

sort_by = "weight"
template = "docs/page.html"

[extra]
toc = true
top = true
+++

> Open mSupply central server is a special site that allows for configurations of subsets of central data. Please see [requirements](/docs/introduction/requirements/#open-msupply-requirements) section for extra info.

Reviewer 2: I didn't know what I was being redirected to - what I should look for on the requirements page.
How about:?

```suggestion
Open mSupply central server is a special site that allows for configuration of subsets of central data. Please see the [requirements](/docs/introduction/requirements/#open-msupply-requirements) section for details on requirements for running open mSupply, and the [Open mSupply central server](docs/introduction/requirements/#open-msupply-central-server) section for details about the central server requirements specifically.
```

Reviewer: Open mSupply

## What is the Open mSupply central server

In essence it's just another Open mSupply instance. However, the differences from a typical remote site are:

- Only one instance of the Open mSupply central server will exist in the Open mSupply system

> - It will be configured by our support staff and needs to be available on the World Wide Web (typically as a cloud server, but can also be hosted in country)

Reviewer 2: :nit the phrase 'in-country' here is a specific thing, and not just 'in a country' which everything is, kinda.
Alternatively, 'hosted locally' or similar.
```suggestion
- It will be configured by our support staff and needs to be available on the World Wide Web (typically as a cloud server, but can also be hosted in-country)
```

- All remote instances of Open mSupply will communicate with Open mSupply central server, as part of the [synchronisation process](/docs/sync/synchronisation/)

> - Allows for configurations of subsets of central data

Reviewer 2: :nit 
```suggestion
- Allows for configuration of subsets of central data
```

## What does it look like?

> It will look very much like any other Open mSupply instance, but you will see a distinct coloured bar at the bottom of the interface which identifies the site as the central server

Reviewer 2: ```suggestion
The central server interface looks very much like any other Open mSupply instance, but you will see a distinct coloured bar at the bottom of the interface which identifies the site as the central server:
```

I would use present tense, not future tense for this. it exists now!

![central server identifier bar!](/docs/introduction/images/central_server_bar_identifier.png)

Similar to the mSupply central server, only one instance of the Open mSupply central server will exist in the Open mSupply system.

## Remote site vs central server

Some operations are only allowed in the Open mSupply central server, if an operation is prohibited on the remote site you will see the following alert

![only allowed on central!](/docs/introduction/images/only_allowed_on_central.png)

Throughout this documentation you will see sections that refer to open mSupply central server functionality, they will look like this and will have a link to this page

[![central server only](/docs/introduction/images/central_server.png '🔗 Available on central server only ')](/docs/introduction/central/#remote-site-vs-central-server)

> ## Configurations and Synchronisations

Reviewer 2: :nit - no need for the plurals
```suggestion
## Configuration and Synchronisation
```

+++
title = "Open mSupply Central Server"
description = "Open mSupply central server functionality"
date = 2022-06-10T11:38:00+00:00
updated = 2022-06-10T11:38:00+00:00
draft = false
weight = 8
sort_by = "weight"
template = "docs/page.html"

[extra]
toc = true
top = true
+++

> Open mSupply central server is a special site that allows for configurations of subset of central data. Please see [requirements](/docs/introduction/requirements/#open-msupply-requirements) section for extra info.

Author: Wanted to capture the version, and the error message and the fact that open mSupply central is required now. Let me know if there is need or better way to do this

> Open mSupply central server is a special site that allows for configurations of subset of central data. Please see [requirements](/docs/introduction/requirements/#open-msupply-requirements) section for extra info.

Reviewer: Hmm will think about it. I'm alright with it staying this way for now (:

## What is open mSupply central server

In essence it's just another open mSupply instance. What makes it difference from a typical remote site ?

- Only one instance of open mSupply central server will exist in open mSupply system
- It will be configured by our support staff and needs to be available on a world wide web (typically as a cloud server, but can also be hosted in country)

> - All remote instance of open mSupply will communicate with open mSupply central server, as part of the synchronisation process

Author: ff32aa7e7c82c0ee3d15614d3077eaedebc6e872, also I think there will be updates to Synchronisation page, either @lache-melvin or I will update the new sync statuses and maybe mention what v6 push and pull refers to

> - All remote instance of open mSupply will communicate with open mSupply central server, as part of the synchronisation process

Reviewer: That would be great. I was just thinking about it (:

- Allows for configurations of subset of central data

## How does open mSupply central server look

It will look very much like any other open mSupply instance, you will see distinct coloured bar at the bottom of interface which identifies the site as central server

![central server identifier bar!](/docs/introduction/images/central_server_bar_identifier.png)

> Similar to mSupply central server, only one instance of open mSupply central server will exist in open mSupply system. 
> Our support staff will be control 

Reviewer: ```suggestion
Similar to the mSupply central server, only one instance of the Open mSupply central server will exist in the Open mSupply system. Contact our support staff for available customisations.
```

Author: Hmm, not 100% sure about this one, i'll remove `Our support staff will be control`, which is unfinished. I am not sure if `Contact our support staff for available customisations.` is quite accurate. I wanted to say that users don't need to be concerned with installing the central server site

5816ade7c1a7f52b3494a826c8c586f87980b036

Reviewer: Ah yeah I wasn't sure where you were going with that XD

## Remote site vs central server

> Some operations are only allowed in open mSupply central server, if operations is prohibited on remote site you will see the following alert

Reviewer: ```suggestion
Some operations are only allowed in the Open mSupply central server, if an operation is prohibited on the remote site you will see the following alert
```

![only allowed on central!](/docs/introduction/images/only_allowed_on_central.png)

Throughout this documentation you will see sections that refer to open mSupply central server functionality, they will look like this and will have a link to this page

> [![central server only](/docs/introduction/images/central_server.png '🔗 Available on central server only ')](/docs/introduction/central/#remote-site-vs-central-server)

Reviewer: Probably don't need to mention this since the places that does mention OMS central server can just link here 🤔 

So far only CCEI has talked about OMS central server and it's just in text, so idk if the image is needed? Or have another sentence in `What does it look like` explaining the icon and text indicator

Author: Like in the pack variant configuration in item catalogue the pack variant editing is visible but only editable on central server, in the docs i thought it would be good to have this link, and consistently every time open mSupply central server is mentioned. Similar in assets when asset can be attached to a store on central, but on remote it will be auto attached to current store. So basically in general docs, we would have areas where things only exists or only possible on central server, and I think it's good to just have a way to flag that to the user.

> OMS central server and it's just in text

Yeah so the point of this link is to indicate it with some visual aid, and consistently across all mentions of central server

Maybe it's good to see it as an example, i'll add in the pack variant docs

How does that sound ?

Reviewer: Sounds good to me (:

## Configurations and Synchronisations

> Open mSupply is configured as another site with a [couple of settings](https://docs.msupply.org.nz/synchronisation:sync_sites#open_msupply_central_server_settings). When open mSupply site goes through syncrhonisation cycle, it will query mSupply central and ask for url where open mSupply central server resides, it will then use this url to synchroniser with open mSupply central site.

Reviewer: ```suggestion
Open mSupply is configured as another site on the mSupply central server with a [couple of extra settings](https://docs.msupply.org.nz/synchronisation:sync_sites#open_msupply_central_server_settings). When the Open mSupply site goes through the synchronisation cycle, it will query the mSupply central server and ask for the URL where the Open mSupply central server resides, it will then use this URL to synchroniser with the Open mSupply central site.
```

> Synchronisation to open mSupply central server is performed via V6 api, and requires a couple of extra steps, as per V6 push and pull in the sync stepper

Reviewer: ```suggestion
Synchronisation to Open mSupply central server is performed via the V6 API, and requires a couple of extra steps, as per the V6 push and pull steps in the sync stepper
```

![sync steps v6!](/docs/introduction/images/sync_steps_v6.png)

## Open mSupply central server data types

> As of 2.0.00

Reviewer: ```suggestion
<div class="note">As of <code>v2.0.00</code></div>
```

> ### Data that is configured in open mSuppy central server

Reviewer: ```suggestion
Data that is configured in Open mSuppy central server:
```

> - Item pack variants
> - Asset catalogue
> - Asset status reasons

Reviewer: ```suggestion
- Item Pack Variants
- Asset Catalogue
- Asset Status Reasons
```

> ### Data that syncrhonises with open mSupply central server

Reviewer: ```suggestion
### Data that synchronises with Open mSupply central server
```
